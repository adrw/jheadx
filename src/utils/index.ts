import * as emoji from "node-emoji"
import chalk from "chalk"
import { execSync } from "child_process"
import yargs = require("yargs")

export const log = console.log
export const logError = (s: string) => log(emoji.emojify(chalk.bold.red(s)))
export const logInfo = (s: string) => log(emoji.emojify(chalk.bold(s)))
export const logDetail = (s: string) => log(emoji.emojify(chalk.dim(s)))

export const jheadCmdFromArgs = () => {
  let args = process.argv
  args.splice(0, 1)
  args[0] = "jhead"
  return args.join(" ")
}

export const runJheadCmd = (cmd: string) => {
  try {
    const result = execSync(cmd, { stdio: "pipe", encoding: "utf-8" })
    console.log(result)
  } catch (e) {
    console.log(e.stdout)
    console.log(e.stderr)
  }
}

export const handleFail = (): void => {
  console.log(yargs.help().version())
  // const cmd = jheadCmdFromArgs()
  // logInfo(`Now running: $ ${cmd}`)
  // runJheadCmd(cmd)
}
