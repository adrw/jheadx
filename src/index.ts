#!/usr/bin/env node

import * as yargs from "yargs"
import { jheadCmdFromArgs, execute, logger } from "./utils"

yargs
  .commandDir("commands")
  .demandCommand(1)
  .option("c", {
    alias: "compareDirectory",
    describe: "source directory (compare)",
    type: "string"
  })
  .option("d", {
    alias: "directory",
    describe: "source directory (master)",
    type: "string"
  })
  .option("s", {
    alias: "start",
    describe: "start time",
    type: "string"
  })
  .option("f", {
    alias: "finish",
    describe: "finish time",
    type: "string"
  })
  .fail((msg, error) => {
    logger.debug(msg)
    logger.error(error)
    const cmd = jheadCmdFromArgs()
    logger.info(`Now running: $ ${cmd}`)
    logger.info(execute(cmd))
  }).argv
