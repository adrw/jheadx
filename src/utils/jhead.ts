const dayjs = require("dayjs")
import { jheadTimeFormat } from "../utils"

export const dumpExifHeader = (path: string) => `jhead -exifmap ${path}`

export const osAgnosticFileDateTime = (path: string) =>
  `${dumpExifHeader(path)} | grep -v Map`

export const dumpFileDateName = (path: string) =>
  `${dumpExifHeader(path)} | sort | tail -n +3 | head -2`

export const dumpExifTimeFileDateName = (path: string) =>
  `${dumpExifHeader(path)} | sort | tail -n +3 | head -3`

export const makeExifSection = (
  path: string,
  outputVerbosityControl: string = "-q"
) => `jhead ${outputVerbosityControl} -mkexif ${path}`

export const setExifTime = (
  newTime: number,
  path: string,
  outputVerbosityControl: string = "-q"
) =>
  `jhead ${outputVerbosityControl} -ts${dayjs(newTime).format(
    jheadTimeFormat
  )} ${path}`

export const setFileTimeToExifTime = (
  path: string,
  outputVerbosityControl: string = "-q"
) => `jhead ${outputVerbosityControl} -ft ${path}`
