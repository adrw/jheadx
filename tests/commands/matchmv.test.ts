import { execute } from "../../src/utils"

describe("jheadx matchmv", () => {
  test("run over real image files", async () => {
    await execute("npm run-script reset")
    expect(
      execute(
        "node dist/index.js matchmv -d ./img/smallTest -m ./img/test -r good"
      )
    ).toMatchSnapshot()
  })
})
