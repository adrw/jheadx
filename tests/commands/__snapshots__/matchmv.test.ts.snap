// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jheadx matchmv run over real image files 1`] = `
"[jheadx][[34mdebug[39m]
Source Directory: ./img/smallTest
[jheadx][[34mdebug[39m]
Match Directory: ./img/test
[jheadx][[34mdebug[39m]
Rename Prefix: good
[jheadx][[34mdebug[39m]
srcFiles: 00006c.jpeg,00009c.jpeg
[jheadx][[34mdebug[39m]
matchFiles: 00006c.jpeg,00007c.jpeg,00008c.jpeg,00009c.jpeg
[jheadx][[34mdebug[39m]
matchFiles: 00007c.jpeg,00008c.jpeg,good00006c.jpeg,good00009c.jpeg
[jheadx][[34mdebug[39m]
Done. üç∫
"
`;
